src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Helpers/Parser/BinderInvocation.cs

```diff
@@ -1,84 +1,72 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics;
 using System.Threading;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Operations;
 using Microsoft.CodeAnalysis;
 
 namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
 {
-    internal sealed record BinderInvocation
+    internal readonly record struct BinderInvocation
     {
-        public IInvocationOperation Operation { get; private set; }
-        public Location? Location { get; private set; }
+        public IInvocationOperation? CandidateOperation { get; private init; }
+        public Location? Location { get; private init; }
 
-        public static BinderInvocation? Create(GeneratorSyntaxContext context, CancellationToken cancellationToken)
+        public static BinderInvocation Create(GeneratorSyntaxContext context, CancellationToken cancellationToken)
         {
-            if (!IsCandidateInvocationExpressionSyntax(context.Node, out InvocationExpressionSyntax? invocationSyntax) ||
-                context.SemanticModel.GetOperation(invocationSyntax, cancellationToken) is not IInvocationOperation operation ||
-                !IsCandidateInvocation(operation))
+            Debug.Assert(IsCandidateSyntaxNode(context.Node));
+            InvocationExpressionSyntax? invocationSyntax = (InvocationExpressionSyntax)context.Node;
+
+            if (context.SemanticModel.GetOperation(invocationSyntax, cancellationToken) is IInvocationOperation operation &&
+                IsCandidateInvocationOperation(operation))
             {
-                return null;
+                return new BinderInvocation()
+                {
+                    CandidateOperation = operation,
+                    Location = invocationSyntax.GetLocation()
+                };
             }
 
-            return new BinderInvocation()
-            {
-                Operation = operation,
-                Location = invocationSyntax.GetLocation()
-            };
+            return default;
```

Comment:
Why not just return `null` as before? Filtering incremental values with `Where` is preferable to having to detect invalid values in the parser layer.